tests/unit/test_gbq.py

```diff
@@ -477,3 +479,29 @@ def test_generate_bq_schema_deprecated():
     with pytest.warns(FutureWarning):
         df = DataFrame([[1, "two"], [3, "four"]])
         gbq.generate_bq_schema(df)
+
+
+def test_load_does_not_modify_schema_arg():
+    # Test of Issue # 277
+    df = DataFrame(
+        {
+            "field1": ["a", "b"],
+            "field2": [1, 2],
+            "field3": [datetime.date(2019, 1, 1), datetime.date(2019, 5, 1)],
+        }
+    )
+    original_schema = [
+        {"name": "field1", "type": "STRING", "mode": "REQUIRED"},
+        {"name": "field2", "type": "INTEGER"},
+        {"name": "field3", "type": "DATE"},
+    ]
+    original_schema_cp = copy.deepcopy(original_schema)
+    gbq.to_gbq(
+        df,
+        "dataset.schematest",
+        project_id="my-project",
+        table_schema=original_schema,
+    )
+    assert [sorted(i.items()) for i in original_schema] == [
+        sorted(i.items()) for i in original_schema_cp
+    ]
```

Comment:
Can we `assert original_schema == original_schema_cp`?

Reply:
Yes, I think that should be okay, but for some reason the original commit/patch isn't doing what it's supposed to and I'm not really sure why.

Here is stepping into the branch locally:

```python
In [3]: original_schema                                                                                                                                      
Out[3]: 
[{'name': 'field1', 'type': 'STRING', 'mode': 'REQUIRED'},
 {'name': 'field2', 'type': 'INTEGER', 'mode': 'NULLABLE'},
 {'name': 'field3', 'type': 'DATE', 'mode': 'NULLABLE'}]
```

So it is still being modified inadvertently despite the `deepcopy()` call.

Reply:
On second check - yes, `assert original_schema == original_schema_cp`.  I was worried about un-orderedness of dictionaries in earlier Python3.  [However](https://docs.python.org/3.4/library/stdtypes.html#dict), 

> Dictionaries compare equal if and only if they have the same (key, value) pairs.